src/orion/storage/track.py

```diff
@@ -0,0 +1,464 @@
+
+        return None
+
+    @property
+    def results(self):
+        """See `~orion.core.worker.trial.Trial`"""
+        return self._results
+
+    @results.setter
+    def results(self, value):
+        """See `~orion.core.worker.trial.Trial`"""
+        self._results = value
+
+    @property
+    def gradient(self):
+        """See `~orion.core.worker.trial.Trial`"""
+        return None
+
+    @property
+    def parents(self):
+        """See `~orion.core.worker.trial.Trial`"""
+        return []
+
+    @property
+    def submit_time(self):
+        """See `~orion.core.worker.trial.Trial`"""
+        return datetime.datetime.utcfromtimestamp(self.storage.metadata.get('submit_time'))
+
+    @property
+    def end_time(self):
+        """See `~orion.core.worker.trial.Trial`"""
+        return datetime.datetime.utcfromtimestamp(self.storage.metadata.get('end_time'))
+
+    @end_time.setter
+    def end_time(self, value):
+        """See `~orion.core.worker.trial.Trial`"""
+        self.storage.metadata['end_time'] = value
+
+
+class Track(BaseStorageProtocol):
+    """Implement a generic protocol to allow Orion to communicate using
+    different storage backend
+
+    Parameters
+    ----------
+    uri: str
+        Track backend to use for storage; the format is as follow
+         `protocol://[username:password@]host1[:port1][,...hostN[:portN]]][/[database][?options]]`
+
+    """
+
+    def __init__(self, uri):
+        self.uri = uri
+        self.options = parse_uri(uri)['query']
+
+        self.client = TrackClient(uri)
+        self.backend = self.client.protocol
+        self.project = None
+        self.group = None
+        self.current_trial = None
+
+        self.objective = self.options.get('objective')
+        assert self.objective is not None, 'An objective should be defined!'
+
+    def create_experiment(self, config):
+        """Insert a new experiment inside the database"""
+        if self.project is None:
+            self.project = self.backend.get_project(Project(name=config['name']))
+
+            if self.project is None:
+                self.project = self.backend.new_project(Project(name=config['name']))
+
+        self.group = self.backend.new_trial_group(
+            TrialGroup(
+                name=str(uuid.uuid4()),
+                project_id=self.project.uid,
+                metadata=to_json(config)
+            )
+        )
+
+        config['_id'] = self.group.uid
+        return config
+
+    def update_experiment(self, experiment, where=None, **kwargs):
+        """Update a the fields of a given trials
+
+        Parameters
+        ----------
+        experiment: Experiment
+            Experiment object to update
+
+        where: Optional[dict]
+            constraint experiment must respect
+
+        **kwargs: dict
+            a dictionary of fields to update
+
+        Returns
+        -------
+        returns true if the underlying storage was updated
+
+        """
+        pass
```

Comment:
experiments cannot be updated for now?

Reply:
tracks create a new experiment every time, the experiments old are never updated

Reply:
How can we support updates then? There is a few fields that can be updated in Oríon, such as `max_trials`.